nixos/modules/services/databases/postgresql.nix

```diff
@@ -361,12 +361,25 @@ in
       };
 
       enableTCPIP = mkOption {
-        type = types.bool;
-        default = false;
+        type = types.nullOr types.bool;
+        default = null;
         description = ''
-          Whether PostgreSQL should listen on all network interfaces.
-          If disabled, the database can only be accessed via its Unix
-          domain socket or via TCP connections to localhost.
+          Whether PostgreSQL should listen on network interfaces.
+          If 'false', the database can only be accessed via its Unix
+          domain socket.
+          A value of 'null' defaults to 'true'.
```

Comment:
Why setting the type to `nullOr bool` when the default is `true` ?

Reply:
The new default is `null`. This has the same behavior as `true` but this makes it possible to show a warning only to users who have this option configured, not to users who left the default values.